### PR TITLE
Create activity - Status step front-end changes

### PR DESCRIPTION
**Pull request by MariluzHerrera**
## Changes in this PR

We recently got some feedback from user research in regards of the way the information was displayed on the Status section of an activity (when creating or editing). The feedback users gave was that it would be more useful to have this data display as radio buttons with hints, instead of a selection box. They also thought that if we display the IATI information for each status on the hints of the buttons, the link to IATI documentation at the top would be no longer needed. 
We also addressed the need of having a dynamic message on this page that tells the user whether they are creating/editing a fund, a project or a programme. 

As per this feedback, the changes on this PR are as follow:

- The content of the label includes now a dynamic value for the level of the current activity
- There is no link to IATI on the help text anymore
- The dropdown selection box has been changed into radio buttons with hints. The hints come from the IATI documentation for each stage of the Status.

When making changes on the view, I addressed a problem with the `yml_to_objects` method. It did not include information needed in this case, such as a `:description`. To solve this, I created a new method on the `codelist_helper.rb` that is specific for status yml data. The reason for this is because the hint text on the radio buttons is given by the `:description` field in `status.yml`, which needed to be included when reading this file but does not need to be included when reading other yml files (eg. currency yml files), as this creates conflicts that result in tests throwing errors.
In addition, I thought that sorting the different status levels by code instead of by name would be more appropriate as they now appear in a more "chronological" order, improving the UX in my opinion.

Some of the tests have changed now to adapt to the new page style, since the way Capybara navigates through radio buttons is slightly different. 

### Before

Screenshot 2020-03-05 at 16 23 35

Screenshot 2020-03-05 at 16 23 48

### After

Screenshot 2020-03-05 at 15 43 24

## Next steps

I noticed that IATI refers to two of the stages with a different name. In our app, stages 3 and 4 are called `Completion` and `Post-completion`, whereas [IATI named them "Finalisation" and "Closed" ](http://reference.iatistandard.org/203/codelists/ActivityStatus/) It is probably worth having a look at this at some point and considering changing them to make sure we are consistent with IATI standards. In addition, I think that descriptions for stages 3 and 4 in our yml file might be the other way around. 

On a different note, I noticed that in the hints of the radio buttons we have the word `activity` instead of a dynamic value for fund, project or programme. Should we consider changing this to a dynamic value again for consistency purposes? 

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x]  Is a changelog entry required? An entry should always be made in CHANGELOG.md, unless this PR is a small tweak which has no impact outside the development team.
- [ ]  Do any environment variables need amending or adding?
- [ ]  Have any changes to the XML been checked with the IATI validator? See XML Validation

### CODE REVIEW

**Comment by jcoglan** [comment]
I just want to add a general comment here that this PR is _excellent_ -- the commits are very well written-up and the PR description covering all the work is similarly detailed and made it easy for me, as someone not familiar with this project, to understand what you've done and why.

Each commit explains why each change needed doing both in terms of user need and technical details, and the PR description gives a lot of context, especially around your thought process for how it fits into a bigger stream of work.

Really good work, well done 🚀

File: spec/support/form_helpers.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def fill_in_activity_form(
     click_button I18n.t("form.activity.submit")
 
     expect(page).to have_content I18n.t("activerecord.attributes.activity.status")
-    expect(page).to have_content "IATI activity status. See IATI for detailed descriptions."
+    expect(page).not_to have_content "IATI activity status. See IATI for detailed descriptions."
```
[the remark this answers is not in the document]
Comment (MariluzHerrera) [reply spec/support/form_helpers.rb]: I had a similar thought. I'll remove this test. Thank you.

File: spec/support/form_helpers.rb [hunk 2]
```diff
@@ -43,8 +43,9 @@ def fill_in_activity_form(
 
     expect(page).to have_content I18n.t("activerecord.attributes.activity.status")
     expect(page).not_to have_content "IATI activity status. See IATI for detailed descriptions."
+    expect(page).to have_content "The activity is currently being implemented"
```
[the remark this answers is not in the document]
Comment (MariluzHerrera) [reply spec/support/form_helpers.rb]: Good point. I agree it won't be the end of the world, so I'll add assertions for the rest of the hints

File: app/helpers/codelist_helper.rb [hunk 3]
```diff
@@ -13,6 +13,13 @@ def yaml_to_objects(entity:, type:, with_empty_item: true)
     objects
   end
 
+  def yaml_to_status_objects(entity:, type:)
```
[the remark this answers is not in the document]
Comment (MariluzHerrera) [reply app/helpers/codelist_helper.rb]: Completely forgot about doing tests for the new method! 😅 I'll give it a go and I'll split the commits so this new method and its tests have its own commit.

**Pull request merged**